spec/features/saml/loa1_sso_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+include SamlAuthHelper
+
+feature 'LOA1 Single Sign On' do
+  context 'First time registration' do
+    before do
+      sign_up_and_set_password
+      visit authnrequest_get
+    end
+
+    it 'prompts user to set up 2FA after confirming email and setting password' do
+      expect(current_path).to eq phone_setup_path
+    end
+
+    it 'prompts the user to confirm phone after setting up 2FA' do
+      fill_in 'Phone', with: '202-555-1212'
+      click_button t('forms.buttons.send_passcode')
+
+      expect(current_path).to eq login_two_factor_path(delivery_method: 'sms')
+    end
+
+    it 'prompts user to acknowledge recovery code before being redirected to the sp' do
+      allow(FeatureManagement).to receive(:prefill_otp_codes?).and_return(true)
+      saml_authn_request = auth_request.create(saml_settings)
+      visit saml_authn_request
+
+      sign_up_and_set_password
+      fill_in 'Phone', with: '202-555-1212'
+      select_sms_delivery
+      enter_2fa_code
+      click_acknowledge_recovery_code
+
+      expect(current_url).to eq saml_authn_request
+    end
+  end
```

Comment:
I know the first 2 `it` blocks here already existed, but what do you think about combining all the expectations into one since this is a slow feature spec? We would reduce the expensive calls to `sign_up_and_set_password` and `visit authnrequest_get` from 3 to 1. Also, we don't really care to test each step individually. Our goal is to make sure that the user gets from point A to point B. Things like `expect(current_path).to eq phone_setup_path` are redundant. We can go straight to `fill_in 'Phone', with: '202-555-1212'`. We only need the one expectation at the end.